py/test/unit/selenium/webdriver/remote/test_remote_connection.py

```diff
@@ -15,21 +15,59 @@
 # specific language governing permissions and limitations
 # under the License.
 
-import pytest
+try:
+    from urllib import parse
+except ImportError:  # above is available in py3+, below is py2.7
+    import urlparse as parse
 
-from selenium.webdriver.remote.remote_connection import RemoteConnection
+from selenium.webdriver.remote.remote_connection import (
+    RemoteConnection,
+    Request
+)
 
 
-def test_basic_auth(mocker):
-    def check(request, timeout):
-        assert request.headers['Authorization'] == 'Basic dXNlcjpwYXNz'
+def test_add_remote_connection_headers_adds_auth_header():
+    url = 'http://user:pass@remote'
+    parsed_url = parse.urlparse(url)
+    cleaned_url = parse.urlunparse((
+        parsed_url.scheme,
+        parsed_url.hostname,
+        parsed_url.path,
+        parsed_url.params,
+        parsed_url.query,
+        parsed_url.fragment)
+    )
+    request = Request(cleaned_url)
```

Comment:
What's the reason for the `cleaned_url`. Can we not just use the original `url` here?